<launch>

    <!-- Start the camera -->
    <node name="camera" pkg="vision" type="camera" output="screen"/>

    <!-- Launch our HSV Filter -->
    <node name="hsv_filter" pkg="vision" type="hsv_filter" output="screen"/>
        <rosparam param="update_frequency"> 5 </rosparam>
        <rosparam param="show_image_window"> true </rosparam>
        <rosparam param="show_calibration_window"> true </rosparam>
    </node>

    <!--
        <!-- Input -->
        <remap from="/robot/camera/raw_image" to="/vision/hsv_input_image"/>

        <!-- Output -->
        <remap from="/robot/vision/hsv_filtered_image" to="/vision/hsv_output_image"/>
    --!>

    <!-- Launch our IPM Filter -->
    <node name="ipm_filter" pkg="vision" type="ipm_filter" output="screen"/>
         <rosparam param="ipm_base_width"> 1 </rosparam>
         <rosparam param="ipm_top_width"> 0.5 </rosparam>
         <rosparam param="ipm_base_displacement"> 0 </rosparam>
         <rosparam param="ipm_top_displacement"> 0 </rosparam>
    </node>

    <!--
        <!-- Input -->
        <remap from="/robot/vision/hsv_filtered_image" to="/vision/ipm_input_image"/>

        <!-- Output -->
        <remap from="/robot/vision/ipm_filtered_image" to="/vision/ipm_output_image"/>
    --!>

    <!-- Start lane detection -->
    <include file="$(find vision)/launch/lane_detection.launch"/>

    <!-- Start the steering driver to control the robot -->
    <include file="$(find drivers)/launch/steering_driver.launch"/>

    <!-- Start the circuit race controller -->
    <node name="lane_follow" pkg="lane_follow" type="lane_follow" output="screen"/>
        <rosparam param="target_x_distance"> 0.5 </rosparam>
        <rosparam param="target_y_distance"> 0.5 </rosparam>
        <rosparam param="angular_speed_multiplier"> 1 </rosparam>
        <rosparam param="linear_speed_multiplier"> 1 </rosparam>
        <rosparam param="theta_multiplier"> 1.75 </rosparam>
        <rosparam param="angular_vel_cap"> 0.75 </rosparam>
        <rosparam param="linear_vel_cap"> 1 </rosparam>
    </node>

</launch>